src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -411,6 +412,20 @@ protected override int MakeSlot(BoundExpression node)
         {
             switch (node.Kind)
             {
+                case BoundKind.Conversion:
+                    {
+                        var conv = (BoundConversion)node;
+                        var operand = conv.Operand;
+                        if (conv.Conversion.Kind == ConversionKind.ExplicitNullable &&
+                            operand.Type?.IsNullableType() == true &&
+                            conv.Type?.IsNullableType() == false)
+                        {
+                            // Explicit conversion of Nullable<T> to T is equivalent to Nullable<T>.Value.
+                            int containingSlot = MakeSlot(operand);
+                            return containingSlot < 0 ? -1 : GetNullableOfTValueSlot(operand.Type, containingSlot);
```

Comment:
>return containingSlot < 0 ? -1 : GetNullableOfTValueSlot(operand.Type, containingSlot); [](start = 28, length = 87)

Can we get here for a conversion from ```Nullable<Type1>``` to ```Type2```?

Reply:
Converting from `Nullable<T1>` to `T2` would be either a `Boxing` conversion or an `ExplicitUserDefined` conversion.

---
In reply to: [238852151](https://github.com/dotnet/roslyn/pull/31499#discussion_r238852151) [](ancestors = 238852151)

Reply:
> Converting from Nullable<T1> to T2 would be either a Boxing conversion or an ExplicitUserDefined conversion.

Are you sure? What about ```int? -> long``` and the like?

---
In reply to: [239155510](https://github.com/dotnet/roslyn/pull/31499#discussion_r239155510) [](ancestors = 239155510,238852151)